### ISSUE TITLE
Improve clarity on the types of Stacks addresses and how they're used

### ISSUE THREAD

**Issue by stjepangolemac** [stacks-network/sbtc#16]
Quoting @jcnelson's [comment](https://github.com/stacks-network/sbtc/pull/8#issuecomment-1658254426):
> StacksAddress does not support p2wpkh or p2sh; it supports p2sh-p2wpkh
and p2sh-p2wsh. This information should be communicated by the new function
names. It was previously communicated in the docs for AddressHashMode.
There are no plans to add p2wsh or p2wsh, because StacksAddress is only
meant to encode Bitcoin address information for addresses that were
supported prior to Stacks 2.0 (StacksAddress inherits these hash modes in
order to provide backwards compatibility)

Looking at the `AddressHashMode` in `blockstack_lib` docs:
> Serialization modes for public keys to addresses. These apply to Stacks addresses, which correspond to legacy Bitcoin addresses – legacy Bitcoin address can be converted directly into a Stacks address, permitting a Bitcoin address to be represented directly on Stacks. These do not apply to Bitcoin segwit addresses.

I'm not completely sure I understand the paragraph above. Let's try to come up with a clear message that we can put into the documentation for `StacksAddress` or constructors of the above mentioned address types.

**Comment by stjepangolemac** [comment]
Tagging @netrome in case you he has some ideas.

**Comment by netrome** [comment]
Wouldn't it be enough to just call the constructors `StacksAddress::p2wpkh_over_p2sh` and `StacksAddress::p2wsh_over_p2sh`?

**Comment by stjepangolemac** [comment]
If you think that's enough then yeah. Maybe we don't really need to go into details in the cargo docs?

**Comment by netrome** [comment]
> If you think that's enough then yeah. Maybe we don't really need to go into details in the cargo docs?

Yeah no I think this is fine. I assume this means the stacks address is equivalent to a Bitcoin `p2wsh_over_p2sh` address, which is pretty well covered by the segwit BIP.

**Comment by friedger** [comment]
Standard Stacks addresses are created using p2pkh (singlesig) p2sh (multisig). Therefore, to me it does not make sense to have the constructors `p2wpkh` and `p2wsh`. 

Also we expect compressed public keys and should handle uncompressed keys.
